recipes/openusd/all/conanfile.py

```diff
@@ -0,0 +1,490 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.53.0"
+
+class OpenUSDConan(ConanFile):
+    name = "openusd"
+    description = "Universal Scene Description"
+    license = "DocumentRef-LICENSE.txt:LicenseRef-Modified-Apache-2.0-License"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://openusd.org/"
+    topics = ("3d", "scene", "usd")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_usd_tools": [True, False],
+        "build_imaging": [True, False],
+        "build_usd_imaging": [True, False],
+        "build_usdview": [True, False],
+        "build_openimageio_plugin": [True, False],
+        "build_opencolorio_plugin": [True, False],
+        "build_embree_plugin": [True, False],
+        "enable_materialx_support": [True, False],
+        "enable_vulkan_support": [True, False],
+        "enable_gl_support": [True, False],
+        "build_gpu_support": [True, False],
+        "enable_ptex_support": [True, False],
+        "enable_openvdb_support": [True, False],
+        "build_renderman_plugin": [True, False],
+        "build_alembic_plugin": [True, False],
+        "enable_hdf5_support": [True, False],
+        "build_draco_plugin": [True, False],
+        "enable_osl_support": [True, False],
+        "build_animx_tests": [True, False],
+        "enable_python_support": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "build_usd_tools": True,
```

Comment:
These are quite a lot of options! We would appreciate if you could submit successful compilation logs some of the most important versions being turned on, as to have traceability over what currently works at this point, thanks!